test/functional/treesitter/language_spec.lua

```diff
@@ -57,8 +57,12 @@ describe('treesitter language API', function()
     local keys, fields, symbols = unpack(exec_lua([[
       local lang = vim.treesitter.language.inspect('c')
       local keys, symbols = {}, {}
-      for k,_ in pairs(lang) do
-        keys[k] = true
+      for k, v in pairs(lang) do
+        if type(v) == 'boolean' then
+          keys[k] = v
+        else
+          keys[k] = true
+        end
```

Comment:
Lua doesn't have ternary, it has `cond and a or b`, however this shouldn't be used if `a` or `b` are booleans since `cond and false or b` -> `b` (`cond` is ignored).

In this case we could write `keys[k] = type(v) ~= 'boolean' or v`, but I think it does obscure the intention of the code.

Reply:
> Lua doesn't have ternary, it has `cond and a or b`

do i have to start typing that out or can I just say "ternary" :)

Reply:
The point is not the terminology; it's the fact that in Lua it's **not** equivalent to a conditional in cases involving booleans. This one of the major footguns we need to guard against.

Reply:
The distinction is quite important and I'm not sure if you knew this. It has been the source of many bugs in core and plugins, including my own.

Reply:
I have proved with personal experience that we should avoid using ternary operations in Lua casually

https://github.com/neovim/neovim/pull/28627